samples/detect.v2beta1.js

```diff
@@ -0,0 +1,848 @@
+      return result.displayName;
+    })
+    .catch(err => {
+      console.error('ERROR:', err);
+    });
+  // [END dialogflow_create_knowledge_base]
+}
+
+function getKnowledgeBase(projectId, knowledgeBaseFullName) {
+  // [START dialogflow_get_knowledge_base]
+  // Imports the Dialogflow client library
+  const dialogflow = require('dialogflow').v2beta1;
+
+  // Instantiate a DialogFlow client.
+  var client = new dialogflow.KnowledgeBasesClient();
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = `your GCLOUD project name, if not specified as env variable e.g. my-Gcloud-Project`;
+  // const knowledgeBaseFullName = `the full path of your knowledge base, e.g my-Gcloud-project/myKnowledgeBase`;
+
+  client
+    .getKnowledgeBase({
+      name: knowledgeBaseFullName,
+    })
+    .then(responses => {
+      const result = responses[0];
+      console.log(`Name: ${result.name}`);
+      console.log(`displayName: ${result.displayName}`);
+    })
+    .catch(err => {
+      console.error('ERROR:', err);
+    });
+  // [END dialogflow_get_knowledge_base]
+}
+
+function listKnowledgeBases(projectId) {
+  // [START dialogflow_list_knowledge_base]
+  // Imports the Dialogflow client library
+  const dialogflow = require('dialogflow').v2beta1;
+
+  // Instantiate a DialogFlow KnowledgeBasesClient.
+  var client = new dialogflow.KnowledgeBasesClient({
+    projectPath: projectId,
+  });
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = `your GCLOUD project name, if not specified as env variable e.g. my-Gcloud-Project`;
+
+  var formattedParent = client.projectPath(projectId);
+
+  client
+    .listKnowledgeBases({parent: formattedParent})
+    .then(responses => {
+      var resources = responses[0];
+      resources.forEach(r => {
+        console.log(`Display name... ${r.displayName}`);
+        console.log(`Knowledge base... ${r.name}`);
+      });
+    })
+    .catch(err => {
+      console.error(err);
+    });
+  // [END dialogflow_list_knowledge_base]
+}
+
+function deleteKnowledgeBase(projectId, knowledgeBaseFullName) {
+  // [START dialogflow_delete_knowledge_base]
+  // Instantiate a DialogFlow client.
+  const dialogflow = require('dialogflow').v2beta1;
+
+  // Instantiate a DialogFlow KnowledgeBasesClient.
+  var client = new dialogflow.KnowledgeBasesClient();
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = `your GCLOUD project name, if not specified as env variable e.g. my-Gcloud-Project`;
+  // const knowledgeBaseFullName = `the full path of your knowledge base, e.g my-Gcloud-project/myKnowledgeBase`;
+
+  client
+    .deleteKnowledgeBase({
+      name: knowledgeBaseFullName,
+    })
+    .then(responses => {
+      const result = responses[0];
+      console.log(`Name: ${result.name}`);
```

Comment:
❗️ **#BLOCKER**

Don't print out the `null` or `undefined` Name please

Instead, follow the canonical: https://github.com/dialogflow/dialogflow-python-client-v2/blob/master/samples/knowledge_base_management.py#L93-L108